### PR TITLE
[FS] Clean-up FWM locking in CCTS

### PR DESCRIPTION
**Pull request by quantumagi**
image

### CODE REVIEW

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        protected object lockObject { get; private set; }
+
+        public LockProtected()
+        {
+            this.lockObject = new object();
+        }
+
+        /// <inheritdoc />
+        public void Synchronous(Action action)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/LockProtected.cs]: Wondering if calling this `Locked` might be better? Or something similar?

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                action.Invoke();
+            }
+        }
+
+        /// <inheritdoc />
+        public T Synchronous<T>(Func<T> action)
+        {
+            lock (this.lockObject)
+            {
+                T res = action.Invoke();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/LockProtected.cs]: You can just go?

```
T res = action();
```

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            this.lockObject = new object();
+        }
+
+        /// <inheritdoc />
+        public void Synchronous(Action action)
+        {
+            lock (this.lockObject)
+            {
+                action.Invoke();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/LockProtected.cs]: You can just call `action()`?

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin/Utilities/LockProtected.cs]: The name is chosen this way to indicate that the code will execute synchronously with other code of the object.

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin/Utilities/LockProtected.cs]: Done

File: src/Stratis.Bitcoin/Utilities/LockProtected.cs (hunk 3 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin/Utilities/LockProtected.cs]: Done

**Comment by codingupastorm** [comment]
Nice, cleaner than what it was. There is a conflict though @quantumagi happy to approve again if you want to fix it.

**Pull request merged**